aws/private/s3_sync.bzl

```diff
@@ -0,0 +1,52 @@
+"Executable rule to release artifacts to an S3 bucket"
+
+_DOC = """\
+Executable rule to copy or sync files to an S3 bucket.
+
+Intended for use with `bazel run`, and with Aspect's Continuous Delivery feature.
+"""
+
+_ATTRS = {
+    "srcs": attr.label_list(
+        doc = "Files to copy to the s3 bucket",
+        allow_files = True,
+        mandatory = True,
+    ),
+    "bucket": attr.label(
```

Comment:
Should we include the other options as well, like `role` and `profile`?

Reply:
Do you think `profile` is typically constant across a team, or do individual engineers usually choose their own values for it?

Reply:
I know that internally we use consistent profiles. But also just exposing the option I think is useful (ie let the customer at least have the option since it's a global config for the CLI)

Reply:
I added a comment showing how to invoke with `--profile` for now. Will leave for a follow-up if we want to add an attribute - it's non-breaking to add but not to remove.
(the shell script would also need some more care to make the default for that flag templated)